tt_metal/fabric/erisc_datamover_builder.cpp

```diff
@@ -196,72 +202,295 @@ FabricEriscDatamoverConfig::FabricEriscDatamoverConfig(std::size_t channel_buffe
+    bool is_dateline_upstream = edm_type == FabricEriscDatamoverType::DatelineUpstream;
 
     if (topology == Topology::Ring) {
-        get_optimal_num_slots(ring_buffer_slot_options, num_sender_buffer_slots, num_receiver_buffer_slots);
+        size_t default_num_sender_buffer_slots;
+        size_t default_num_receiver_buffer_slots;
+        // get the default buffer slots
+        get_optimal_num_slots(
+            ring_buffer_slot_options,
+            this->num_used_sender_channels,
+            this->num_used_receiver_channels,
+            default_num_sender_buffer_slots,
+            default_num_receiver_buffer_slots);
+        // get the dateline buffer slots
+        size_t dateline_num_sender_buffer_slots;
+        size_t dateline_num_receiver_buffer_slots;
+        get_optimal_num_slots(
+            ring_buffer_slot_options_dateline,
+            this->num_used_sender_channels - 1,
+            this->num_used_receiver_channels - 1,
+            dateline_num_sender_buffer_slots,
+            dateline_num_receiver_buffer_slots);
+
+        // set default num_sender_buffer_slots
+        for (size_t i = 0; i < this->num_used_sender_channels; ++i) {
+            num_sender_buffer_slots[i] = default_num_sender_buffer_slots;
+        }
+        // set default remote_sender_buffer_slots
+        for (size_t i = 0; i < this->num_used_sender_channels; ++i) {
+            num_remote_sender_buffer_slots[i] = default_num_sender_buffer_slots;
+        }
+        // set default num_receiver_buffer_slots
+        for (uint32_t i = 0; i < this->num_used_receiver_channels; i++) {
+            num_receiver_buffer_slots[i] = default_num_receiver_buffer_slots;
+        }
+        // set default remote_receiver_buffer_slots
+        for (uint32_t i = 0; i < this->num_used_receiver_channels; i++) {
+            num_remote_receiver_buffer_slots[i] = default_num_receiver_buffer_slots;
+        }
+        // set default downstream_num_sender_buffer_slots
+        downstream_num_sender_buffer_slots = default_num_sender_buffer_slots;
+        this->remote_receiver_channel_num_buffers = default_num_receiver_buffer_slots;
+
+        size_t dateline_upstream_num_receiver_buffer_slots = has_ring_loopback ? default_num_receiver_buffer_slots
+                                                             : en_dateline_upstream_receiver_extra_buffer
+                                                                 ? dateline_num_receiver_buffer_slots
+                                                                 : default_num_receiver_buffer_slots;
+        size_t dateline_upstream_num_sender_buffer_slots = has_ring_loopback ? default_num_sender_buffer_slots
+                                                           : en_dateline_upstream_sender_extra_buffer
+                                                               ? dateline_num_sender_buffer_slots
+                                                               : default_num_sender_buffer_slots;
```

Comment:
I personally prefer normal `if` instead of nested ternary operator

```c++
size_t dateline_upstream_num_receiver_buffer_slots = default_num_receiver_buffer_slots
size_t dateline_upstream_num_sender_buffer_slots = default_num_sender_buffer_slots
if (!has_ring_loopback) {
.... // ternary operator or if
}

```